mariadb_kernel/kernel.py

```diff
@@ -141,10 +143,52 @@ def do_execute(
 
         return rv
 
-    def do_shutdown(self, restart):
-        self.mariadb_client.stop()
+    def num_connected_clients(self):
+        sql_query = "select count(id) from information_schema.processlist;"
+        # If there is any errors we raise
+        # (not much we can do if there is a error)
+        result_html = self.mariadb_client.run_statement(code=sql_query)
+        if self.mariadb_client.iserror():
+            raise Exception
+        df = pandas.read_html(result_html)
+        num_clients = int(df[0]["count(id)"][0])
+
+        return num_clients
+
+    def kill_server(self):
         if self.mariadb_server and self.mariadb_server.is_up():
+            self.log.info(f"Stopping (own) MariaDB server")
             self.mariadb_server.stop()
+        elif not self.mariadb_server:
+            pidfile = self.client_config.get_server_pidfile()
+            try:
+                with open(pidfile, "r") as f:
+                    pid = int(f.read())
+            except Exception:
```

Comment:
should be changed to something like `RuntimeError`. I've just tried debugging why the thing does not work (I forgot to change "count(id)" to "Value") and except here was catching a dataframe error.

Reply:
I wanted to catch all exception as it is a bit dangerous to interrupt the code when deleting/shutting down.  Maybe a middle way could be to give some better debug output?

Reply:
logging the exception too might be a reasonable solution, if somehow during development Exception catches a syntax error, at least you get a clue of what is going on.